### PR TITLE
[WIP] Add support for quoted tweets - needs styling

### PR DESCRIPTION
**Pull request by AlexVanderbist**
- Started listening to the Twitter User stream for `quoted_tweet` events (backend)
- Added `quote` and `hasQuote` properties to the Tweet class (frontend)
- Added HTML structure for quoted tweet in `Twitter` component
- Added front-end tests

### CODE REVIEW

**Comment by freekmurze** [comment]
Thanks!

File: app/Components/Twitter/ListenForMentions.php [hunk 1]
```diff
@@ -37,5 +37,15 @@ public function handle()
                 event(new Mentioned($tweetProperties));
             })
             ->startListening();
+
+        app(TwitterStreamingApi::class)
+            ->userStream()
+            ->onEvent(function(array $event) {
+                if (isset($event['event']) && $event['event'] === 'quoted_tweet') {
+                    dump($event);
```
Comment (freekmurze) [inline app/Components/Twitter/ListenForMentions.php]: remove this

File: resources/assets/js/services/twitter/Tweet.js [hunk 2]
```diff
@@ -2,10 +2,14 @@ import { get } from 'lodash';
 import moment from 'moment';
 import twemoji from 'twemoji';
 
-export default class {
+class Tweet {
```
Comment (freekmurze) [inline resources/assets/js/services/twitter/Tweet.js]: why this change?

File: app/Components/Twitter/ListenForMentions.php (hunk 1 above)
Comment (AlexVanderbist) [inline app/Components/Twitter/ListenForMentions.php]: Oops, my bad!

File: resources/assets/js/services/twitter/Tweet.js (hunk 2 above)
Comment (AlexVanderbist) [inline resources/assets/js/services/twitter/Tweet.js]: To be able to create a new instance of the Tweet class for a quote (in the class' constructor):
```js
this.quotedTweet = new Tweet(this.tweetProperties.quoted_status);
```
I couldn't find another way to create a new instance of an unnamed class

Comment (freekmurze) [inline resources/assets/js/services/twitter/Tweet.js]: ok 👍
